internal/app/machined/pkg/controllers/block/user_disk_config.go

```diff
@@ -123,6 +123,7 @@ func (ctrl *UserDiskConfigController) Run(ctx context.Context, r controller.Runt
 								Match: partitionIdxMatch(resolvedDevicePath, idx+1),
 							}
 
+							// TODO: does this need a SELinux label?
```

Comment:
This is unclear, user disks are totally not something that Talos should touch or use for system purposes.

Reply:
Would be easier, as relabeling by FS boundaries is easier than file-discretely. We have to actually handle relabeling of disk-backed data on updates, we might need to think a bit about this, since what if we update from 1.8 in which all the files have been already created and won't gain labels by type transitions

Reply:
Any other ideas what to label discretely?

Reply:
I'd say there are three potential candidates:

* etcd
* containerd state (might be hard, as it contains also container's rootfs)
* kubelet state

We also have some overlay mounts from `/var` to `/etc`, idk if it makes sense to protect their upper dir in `/var`

Reply:
All these make sense, thanks, will do